embedded-uart-common/sensirion_shdlc.c

```diff
@@ -188,11 +188,9 @@ int16_t sensirion_shdlc_rx(uint8_t max_data_len,
         return SENSIRION_SHDLC_ERR_ENCODING_ERROR;
 
     crc = rx_frame[i++];
-    if (sensirion_shdlc_check_unstuff(crc)) {
+    if (sensirion_shdlc_check_unstuff(crc))
         crc = sensirion_shdlc_unstuff_byte(rx_frame[i++]);
-        if (i >= len)
-            return SENSIRION_SHDLC_ERR_MISSING_STOP;
```

Comment:
What is with this error condition? Can it just be removed?

Reply:
It's checking if the pointer moves over the end of the frame. In this case we can remove it because on the CRC check follows which is guaranteed to stay in bound - in that case, however, we could still accept a frame that's missing the final frame delimiter, but this is checked again at the end (L200). So basically this eliminates a duplicate check at the cost of performing the CRC check despite a missing frame stop (that we now don't detect earlier on).